### PR TITLE
⭐️New: Add `vue/component-name-in-template-casing`

### PR DESCRIPTION
**Pull request by trollepierre**
Add modification asked during review of 
https://github.com/vuejs/eslint-plugin-vue/pull/397

### CODE REVIEW

File: lib/rules/component-name-in-template-casing.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'enforce specific casing for the component naming style in template',
+      category: undefined,
+      url: 'https://github.com/vuejs/eslint-plugin-vue/blob/v4.3.0/docs/rules/html-element-name-casing.md'
```
Comment (shroudedcode) [inline lib/rules/component-name-in-template-casing.js]: This has to be changed to `'https://github.com/vuejs/eslint-plugin-vue/blob/v4.3.0/docs/rules/component-name-in-template-casing.md'`.

**Comment by shroudedcode** [comment]
@trollepierre You forgot to rename the documentation file to `html-element-name-casing.md`.

**Comment by trollepierre** [comment]
I made the change ^^ Thanks guys!

**Comment by shroudedcode** [comment]
@trollepierre The build is failing. `AssertionError [ERR_ASSERTION]: Should have 1 error but had 0` for three of the tests. Seems like **invalid** element names are **not** reported correctly in these cases.

**Comment by trollepierre** [comment]
Thanks @shroudedcode ! I have seen it before but I couldn't fix it before now.

**Comment by trollepierre** [comment]
Can we merge this PR?

**Comment by shroudedcode** [comment]
@trollepierre I would say yes, but [there's still no consensus on the rule name and the default casing](https://github.com/vuejs/eslint-plugin-vue/pull/397#issuecomment-393883049) ...

**Comment by michalsnik** [comment]
Thanks for pushing this topic forward @trollepierre I have to however close this PR due to the final consensus in #397

**Comment by trollepierre** [comment]
No problem! So when will be the next release?

**Pull request closed without merging**